client/van_serviceinterface_update_test.go

```diff
@@ -0,0 +1,394 @@
+		Aggregate:    "",
+	})
+	assert.Error(t, err, "Skupper not initialised in van-serviceinterface-update")
+
+	err = cli.VanRouterCreate(ctx, types.VanSiteConfig{
+		Spec: types.VanSiteConfigSpec{
+			SkupperName:       "skupper",
+			IsEdge:            false,
+			EnableController:  true,
+			EnableServiceSync: true,
+			EnableConsole:     false,
+			AuthMode:          "",
+			User:              "",
+			Password:          "",
+			ClusterLocal:      true,
+		},
+	})
+	assert.Assert(t, err)
+
+	// create three service definitions
+	time.Sleep(time.Second * 1)
+	err = cli.VanServiceInterfaceCreate(ctx, &types.ServiceInterface{
+		Address:      "tcp-go-echo",
+		Protocol:     "tcp",
+		Port:         9090,
+		EventChannel: false,
+		Aggregate:    "",
+	})
+	assert.Assert(t, err)
+
+	err = cli.VanServiceInterfaceCreate(ctx, &types.ServiceInterface{
+		Address:      "nginx",
+		Protocol:     "http",
+		Port:         8080,
+		EventChannel: false,
+		Aggregate:    "",
+	})
+	assert.Assert(t, err)
+
+	err = cli.VanServiceInterfaceCreate(ctx, &types.ServiceInterface{
+		Address:      "tcp-go-echo-ss",
+		Protocol:     "tcp",
+		Port:         9090,
+		EventChannel: false,
+		Aggregate:    "",
+	})
+	assert.Assert(t, err)
+
+	time.Sleep(time.Second * 1)
+
+	// bind services to targets
+	// TODO: could range on list if target type was not needed for bind
+	si, err := cli.VanServiceInterfaceInspect(ctx, "tcp-go-echo")
+	assert.Assert(t, err)
+	err = cli.VanServiceInterfaceBind(ctx, si, "deployment", "tcp-go-echo", "tcp", 9090)
+	assert.Assert(t, err)
+
+	si, err = cli.VanServiceInterfaceInspect(ctx, "tcp-go-echo-ss")
+	assert.Assert(t, err)
+	err = cli.VanServiceInterfaceBind(ctx, si, "statefulset", "tcp-go-echo-ss", "tcp", 9090)
+	assert.Assert(t, err)
+
+	si, err = cli.VanServiceInterfaceInspect(ctx, "nginx")
+	assert.Assert(t, err)
+	// bad bind
+	err = cli.VanServiceInterfaceBind(ctx, si, "deployment", "nginx2", "http", 8080)
+	assert.Error(t, err, "Could not read deployment nginx2: deployments.apps \"nginx2\" not found")
+	// good bind
+	err = cli.VanServiceInterfaceBind(ctx, si, "deployment", "nginx", "http", 8080)
+	assert.Assert(t, err)
+
+	items, err := cli.VanServiceInterfaceList(ctx)
+	assert.Assert(t, err)
+	assert.Equal(t, len(items), 3)
+
+	if *clusterRun {
+		// this delay is for service-controller to update
+		time.Sleep(time.Second * 30)
+		if diff := cmp.Diff(svcsExpected, svcsFound, trans); diff != "" {
+			t.Errorf("TestVanServiceInterfaceUpdate services mismatch (-want +got):\n%s", diff)
+		}
+	}
+
+	for _, c := range testcases {
+		ctx, cancel := context.WithCancel(ctx)
+		defer cancel()
+
+		si, err := cli.VanServiceInterfaceInspect(ctx, c.name)
+		assert.Check(t, err, c.doc)
+
+		if c.port != 0 {
+			si.Port = c.port
+		}
+		if c.eventChannel != si.EventChannel {
+			si.EventChannel = c.eventChannel
+		}
+		if c.aggregate != si.Aggregate {
+			si.Aggregate = c.aggregate
+		}
+		err = cli.VanServiceInterfaceUpdate(ctx, si)
```

Comment:
After updating, isn't it missing to test for the particular attribut updated? i.e: port modified from 9090 to 9091?